Reviews backend

Oleg-Lopes opened the pull request:
reviews/{id}/edit to approve/edit/delete review
reviews/{id}/comments to approve/edit/delete comments of that review
only admin can do that

only logged in users can leave comments and reviews below movies/series

Commit:
add routes for reviews

Commit:
create and store done

- can see link to create review only if logged in
- can create review only if logged in
- can create review only on movie/series
- displaying review with title, username and body

Commit:
redirect back to title after creating a review

Commit:
Merge branch 'dev' into reviews-backend

Commit:
logged user can add comments

Commit:
only approved reviews and comments are visible

Commit:
reviews can be updated (edited)

Commit:
delete review and all of it comments

Commit:
delete and update comments

SebastianGerS commented:
Looking at this now (because I can't sleep :P)

The reason the styling is not loaded properly is because you have renamed the the routes for the series/episodes/movies to be prefixed with the name of said element.

This makes sens. but the way I have setup which stylesheets is loaded works by looking at the routename and then loads in the style sheet with the same name. info about this is in the reademe and the code for it is in the header.blade. 
because all three routes uses the same styling so I named them all title (for @show) and edit (for @edit)  and this corresponds to the css files where the styling for the pages are. we could make three different stylesheets but that seams a bit redundant.

but I can see that you are using the routename for a different purpose in your reviewscontroller, namely to redirect to one of two routs by giving them different names, but actually i think you could just use  `back()->withInput();`  
It takes the user back to the route where the request was sent from
see [https://laravel.com/docs/5.5/redirects] for more info.

also the stars column (which I don't remember from earlier) seams redundant we already have a rating system and it seams weird to be able to rate the movie (function for this is in the titlescontroller@rate) both in the review and on the movie it self.

SebastianGerS commented:
but well done overall!
And i'm guessing you added the start column because of that they where there in the design,
meaby you could try to move the main part of my ratings function in titlecontroller to the databasehelpers trait and reuse it in the reviewscontroller that should to the trick

Commit:
reverted to corect naming scheem for routenames

Commit:
move rating functionality to dabasehelpers to make it possible to combine that functionality with other routes

Commit:
bugfix for showcontent function

	* wrapt buttons conserning comment in an if statement to check
	if the buttons are renderd befortring to set eventlisteners on
	them

	explenation: button with comment prefix in id only renders if
	theres an review on the title curently beeing viewed, so when
	there where o comments the function could not find eny elements
	and an error was thrown whiche made all buttons in that function
	unresponvi

Commit:
removed stars column

Commit:
removed fillable stars column

Commit:
bugfix in comment/review Controller

	* remove refferences to Series/Movies prefixed routenames
	* simplify redirection with use of url()->previous()
	* removed stars from ITEMCOLUMS constant
	* removed start as column in review creation
	* add func to add record in title_user_rating

Commit:
remove referenses to stars on review and fix incrorect id for
	cancel-comment button

SebastianGerS commented:
I have fixed all the bugs i found in this bransch so will merge this now.
